src/main/java/org/openrewrite/java/spring/boot3/MigrateWebMvcTagsToObservationConvention.java

```diff
@@ -0,0 +1,285 @@
+    private static final String KEYVALUE_FQ = "io.micrometer.common.KeyValue";
+    private static final String HTTPSERVLETREQUEST_FQ = "jakarta.servlet.http.HttpServletRequest";
+    private static final String HTTPSERVLETRESPONSE_FQ = "jakarta.servlet.http.HttpServletResponse";
+    private static final String TAGS_FQ = "io.micrometer.core.instrument.Tags";
+    private static final String TAG_FQ = "io.micrometer.core.instrument.Tag";
+    private static final MethodMatcher TAGS_AND_STRING_STRING = new MethodMatcher("io.micrometer.core.instrument.Tags and(java.lang.String, java.lang.String)");
+    private static final MethodMatcher TAGS_AND_STRING_ARRAY = new MethodMatcher("io.micrometer.core.instrument.Tags and(java.lang.String[])");
+    private static final MethodMatcher TAGS_AND_TAG_ARRAY = new MethodMatcher("io.micrometer.core.instrument.Tags and(io.micrometer.core.instrument.Tag[])");
+    private static final MethodMatcher TAGS_AND_TAG_ITERABLE = new MethodMatcher("io.micrometer.core.instrument.Tags and(java.lang.Iterable)");
+    private static final MethodMatcher TAGS_OF_STRING_STRING = new MethodMatcher("io.micrometer.core.instrument.Tags of(java.lang.String, java.lang.String)");
+    private static final MethodMatcher TAGS_OF_STRING_ARRAY = new MethodMatcher("io.micrometer.core.instrument.Tags of(java.lang.String[])");
+    private static final MethodMatcher TAGS_OF_TAG_ARRAY = new MethodMatcher("io.micrometer.core.instrument.Tags of(io.micrometer.core.instrument.Tag[])");
+    private static final MethodMatcher TAGS_OF_TAG_ITERABLE = new MethodMatcher("io.micrometer.core.instrument.Tags of(java.lang.Iterable)");
+    private static final MethodMatcher TAGS_OF_ANY = new MethodMatcher("io.micrometer.core.instrument.Tags of(..)");
+    private static final MethodMatcher TAG_OF = new MethodMatcher("io.micrometer.core.instrument.Tag of(java.lang.String, java.lang.String)");
+
+    @Override
+    public String getDisplayName() {
+        return "Migrate `WebMvcTagsProvider` to `DefaultServerRequestObservationConvention`";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Migrate `WebMvcTagsProvider` to `DefaultServerRequestObservationConvention` as part of Spring Boot 3.2 removals.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(new UsesType<>(WEBMVCTAGSPROVIDER_FQ, true), new JavaVisitor<ExecutionContext>() {
+            @Override
+            public J.ClassDeclaration visitClassDeclaration(J.ClassDeclaration classDecl, ExecutionContext ctx) {
+                List<Statement> getTagsBodyStatements = new ArrayList<>();
+                for (Statement stmt : classDecl.getBody().getStatements()) {
+                    if (stmt instanceof J.MethodDeclaration) {
+                        J.MethodDeclaration md = (J.MethodDeclaration) stmt;
+                        if (md.getSimpleName().equals("getTags") && md.getBody() != null) {
```

Comment:
Not sure if we should use a method matcher to be on the safe side. Moving the `md` variable to the top-level scope of this method would be good, so that we can use it later on.